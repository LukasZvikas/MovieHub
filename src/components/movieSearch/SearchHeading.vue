
<template>
  <div>
    <div class="search-text-heading" data-test="text-heading">{{heading}}</div>
    <div class="search-text-body" data-test="text-body">{{textBody}}</div>
  </div>
</template>

<script>
export default {
  props: {
    heading: String,
    textBody: String
  }
};
</script>

<style>
.search-text-heading {
  color: #fff;
  font-size: 3rem;
  text-align: center;
}

.search-text-body {
  color: #fff;
  font-size: 1.6rem;
  text-align: center;
  margin: 1.6rem 5rem;
}
</style>
