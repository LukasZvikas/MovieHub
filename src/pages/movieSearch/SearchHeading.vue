<template>
  <div class="heading-2 p-2 heading-bg-color">
    <div
      class="text-center font-weight-bold pb-1 heading-1"
      data-test="movie-search-heading"
      style="padding-top: 0!important"
    >{{heading}}</div>
    <div class="search-text-body" data-test="movie-search-body">{{textBody}}</div>
  </div>
</template>

<script>
export default {
  props: {
    heading: String,
    textBody: String
  }
};
</script>

<style>
.search-text-heading {
  color: #fff;
  font-size: 3rem;
  text-align: center;
}

.heading-bg-color {
  background-color: rgba(25, 20, 20, 0.7);
}

.search-text-body {
  color: #fff;
  font-size: 1.6rem;
  text-align: center;
}
</style>
