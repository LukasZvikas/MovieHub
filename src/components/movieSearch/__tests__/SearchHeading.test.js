import { shallowMount, mount } from "@vue/test-utils";
import SearchHeading from "../SearchHeading";

describe("SearchHeading", () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      heading: "Discover movies here!",
      textBody:
        "Our database stores pretty much every movie ever made. Give it a shot and you will see!"
    };
    wrapper = shallowMount(SearchHeading, {
      propsData: { ...props }
    });
  });

  it("has a correct heading text", () => {
    expect(wrapper.find('[data-test="text-heading"]').text()).toBe(
      props.heading
    );
  });

  it("has a correct textBody text", () => {
    expect(wrapper.find('[data-test="text-body"]').text()).toBe(props.textBody);
  });

  it("renders content correctly", () => {
    const wrapper = mount(SearchHeading, {
      propsData: { ...props }
    });
    expect(wrapper.vm.$el).toMatchSnapshot();
  });
});
