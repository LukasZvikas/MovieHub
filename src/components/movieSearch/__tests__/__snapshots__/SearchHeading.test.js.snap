// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeading renders content correctly 1`] = `
<div>
  <div
    class="search-text-heading"
    data-test="text-heading"
  >
    Discover movies here!
  </div>
   
  <div
    class="search-text-body"
    data-test="text-body"
  >
    Our database stores pretty much every movie ever made. Give it a shot and you will see!
  </div>
</div>
`;
